software/spmd/bsg_cuda_lite_runtime/matrix_mul_shared_mem_hard/kernel_matrix_mul_shared_mem_hard.cpp

```diff
@@ -0,0 +1,244 @@
+// * This kernel performs blocked matrix multiplication using 
+//   tile group shared memory. Subblocks of input matrices are
+//   loaded into tile group shared memory by all tiles, and 
+//   accesses for matrix multiplication are done to the 
+//   lower-latency tile group shared meomry instead of the long-
+//   latency DRAM.
+// * This code uses the hardware tile group shared memory 
+//   in conjunction with it's library bsg_shared_mem.hpp
+// * Tile group dimensions are fixed at 4x4.
+
+// TEMPLATE_TG_DIM_X/Y must be defined before bsg_manycore.h is
+// included. bsg_tiles_X and bsg_tiles_Y must also be defined for
+// legacy reasons, but they are deprecated.
+
+
+#define TEMPLATE_TG_DIM_X 4
+#define TEMPLATE_TG_DIM_Y 4
+#define TEMPLATE_BLOCK_SIZE_X  64
+#define TEMPLATE_BLOCK_SIZE_Y  64
+#define TEMPLATE_SUBBLOCK_SIZE 32
+#define TEMPLATE_STRIPE_SIZE   1
+#define bsg_tiles_X TEMPLATE_TG_DIM_X
+#define bsg_tiles_Y TEMPLATE_TG_DIM_Y
```

Comment:
The CUDA makefiles in replicant specify a 2x2, but this is a 4x4.

Reply:
Yes I'm aware, I'm using pre-fixed 4x4 dimensions for this specific kernel, because for 2x2 tile groups the shared memory would be too small. And I didn't want to change the bsg_replicant infrastructure since the tile group dimensions there are 2x2. The replicant Makefile specifies a 2x2 but the 4x4 inside the kernel overrides that. Good catch though!

Reply:
Is this resolved?

Reply:
I don't know - best to check if the source in replicant and here agree.